lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java

```diff
@@ -395,10 +396,8 @@ public void deleteClassification(String classificationId)
     }
 
     private boolean isReferentialIntegrityConstraintViolation(PersistenceException e) {
-        return ((e.getCause().getClass().getName().equals("org.h2.jdbc.JdbcSQLException")
-            && e.getMessage().contains("23503"))
-            || (e.getCause() instanceof SQLIntegrityConstraintViolationException
-                && e.getMessage().contains("-532")));
+        return e.getCause() instanceof SQLException && ((SQLException) e.getCause()).getSQLState().equals("23503")
```

Comment:
ok, this is tested with all databases?

Reply:
so far with postgres & h2. db2 is coming later today